### PR TITLE
Homepage redesign 2023

### PR DESCRIPTION
**Pull request by reemhamz**
## One-line summary
Redesigning the homepage!

[Figma file](https://www.figma.com/file/gGW6xUpetI6wjsnccWWXvV/mozilla.org-refresh?type=design&node-id=422%3A975&mode=design&t=YEGitsPA6i3t3Yvw-1) (also includes copy).

This new homepage will be localized for all locales and will start to replace our current rest-of-world homepage \o/.'

‼️ **Note:** The new homepage will only start showing for locales once they've been translated. 

VPN Promo switch PR: https://github.com/mozmeao/www-config/pull/527

## Significant changes and points to review
- [x] Going to have a VPN promo to start out the re-launch of this page
- [x] VPN promo to be part of a variant experiment, where the alternative is to have no banner at all:
- [x] Remove French and German homepages
- [x] Update functional tests to reflect new changes
- [x] 50% VPN promo banner
- [x] 50% no VPN promo banner

## Testing
- http://localhost:8000/en-US/?v=1 (no promo banner)
- http://localhost:8000/en-US/?v=2 (w/ promo banner)
- https://www-demo6.allizom.org/en-US/?v=1 (no promo banner)
- https://www-demo6.allizom.org/en-US/?v=2 (w/ promo banner)
- Integration tests [have passed](https://github.com/mozilla/bedrock/actions/runs/5829952104/workflow)
- Add `HOME_VPN_FEATURE_EXPERIMENT=True` to `.env` file to test VPN promo variation

To test functional tests, run: 
```
py.test --base-url http://localhost:8000 --driver Firefox --html tests/functional/results.html tests/functional/test_home.py
```

### CODE REVIEW

File: l10n/en/mozorg/home-old.ftl
Comment (alexgibson) [inline l10n/en/mozorg/home-old.ftl:5]: Heads up: renaming an existing FTL file will break translations for the old home page. It's probably better to create a new FTL file for the redesigned home page, and leave the old FTL file intact

Comment (reemhamz) [reply l10n/en/mozorg/home-old.ftl:5]: I have that in my notes to fix up! :) thanks for the sharp eyes here

File: bedrock/mozorg/views.py
Comment (stevejalim) [inline bedrock/mozorg/views.py:138]: Is this to deliberately disable the old en homepages (Contentful or templated)? If so, might as well just drop the entire if clause and start with `if locale == "de":`

Comment (reemhamz) [reply bedrock/mozorg/views.py:138]: I made it a typo when I first started working on this PR so the rest-of-world homepage would appear instead of Contentful. It was a janky hacky way of getting things rolling without messing with this Python file too much. I'm planning on cleaning all this up before having it reviewed

**Comment by reemhamz** [comment]
**Note:** Above unit test failed ([reference](https://github.com/mozilla/bedrock/actions/runs/5711346004/job/15472838579?pr=13447#step:4:4827)) because it's looking for `home-contentful.html` and I removed it in the recent commit. Just confirming a few things w/ the team regarding the test script before I fix it.

File: bedrock/mozorg/views.py
Comment (reemhamz) [inline bedrock/mozorg/views.py]: **Note:** Removed this view because we're no longer going to have dedicated homepages for French and German locales. This homepage will be global.

File: bedrock/mozorg/templates/mozorg/home/home-old.html
Comment (alexgibson) [inline bedrock/mozorg/templates/mozorg/home/home-old.html:191]: This is throwing an error on the old home page since `high_res_img()` no longer exists

File: media/css/mozorg/home/home-new.scss
Comment (alexgibson) [inline media/css/mozorg/home/home-new.scss:231]: Issue: bidi()
```suggestion
                @include bidi(((text-align, left, right),));
```

File: media/css/mozorg/home/home-new.scss
Comment (alexgibson) [inline media/css/mozorg/home/home-new.scss:237]: Issue: bidi()
```suggestion
            @include bidi(((margin-left, $layout-lg, margin-right, 0),));
```

File: bedrock/mozorg/templates/mozorg/home/home-contentful.html
Comment (alexgibson) [inline bedrock/mozorg/templates/mozorg/home/home-contentful.html:9]: Question: can all these old macros be removed?

**Review by alexgibson — CHANGES_REQUESTED** [review]
This new home page looks great and is a solid improvement over what we've had for years now, nice work!

The new page itself looks great, only a few minor nits. There does seem to be quite a bit of left over assets from the old pages that could do with tidying up / removing (images, promos, CSS bundles etc), but we could always do that in a followup if easier.

I also asked questions around things this new home page is losing, like the fundraising banner and the Firefox app store banner. Are those supposed to be no longer a thing were showing?

File: bedrock/mozorg/templates/mozorg/home/home-contentful.html
Comment (reemhamz) [reply bedrock/mozorg/templates/mozorg/home/home-contentful.html:9]: I'm thinking we keep the `home/includes/macros.html` file for a bit in case we may need to add them into our new homepage later on?

File: bedrock/mozorg/templates/mozorg/home/home-contentful.html
Comment (reemhamz) [reply bedrock/mozorg/templates/mozorg/home/home-contentful.html:48]: Good callout -- I'll do that :) Let me just double-check with @slightlyoffbeat if we should keep the Pocket promo in case it's still being used.

File: bedrock/mozorg/templates/mozorg/home/home-contentful.html
Comment (alexgibson) [reply bedrock/mozorg/templates/mozorg/home/home-contentful.html:9]: Up to you 👍  The design for these macros is pretty old though, so I'd be surprised if we incorporated them back into the new design

Comment (reemhamz) [reply bedrock/mozorg/templates/mozorg/home/home-contentful.html:9]: Oh fair, well I guess in that case we should be fine to remove. If we end up needing them, I'm sure we can fetch 'em from our git history

File: bedrock/mozorg/templates/mozorg/home/home-contentful.html
Comment (reemhamz) [reply bedrock/mozorg/templates/mozorg/home/home-contentful.html:13]: In this case, should I make the `en-US` and `en-GB` locale versions of the page title hard-coded in the HTML file and not the Fluent file?

Comment (reemhamz) [reply bedrock/mozorg/templates/mozorg/home/home-contentful.html:13]: Ah just checked with other files and that seems the way to go forward. I've added this code to replace the origin al `page_title` codeblock:

```
{%- block page_title_full -%}
  {%- if LANG == 'en-US' -%}
    Internet for people, not profit — Mozilla (US)
  {%- elif LANG == 'en-GB' -%}
    Internet for people, not profit — Mozilla (UK)
  {%- else -%}
    {{ ftl('home-internet-for-people-not-profit') }}
  {%- endif -%}
{%- endblock -%}
{% block page_title_suffix %}{% endblock %}
```

File: bedrock/mozorg/templates/mozorg/home/home-new.html
Comment (reemhamz) [reply bedrock/mozorg/templates/mozorg/home/home-new.html:145]: I was only given these low-res images, so I'm going to have to ask studio for higher quality images

File: bedrock/mozorg/templates/mozorg/home/includes/featured-vpn.html
Comment (reemhamz) [inline bedrock/mozorg/templates/mozorg/home/includes/featured-vpn.html:11]: I moved this promo's CSS bundle into its HTML file, because it was interfering with banner conditionals when it was placed in the `home-new.html` file.

Comment (alexgibson) [reply bedrock/mozorg/templates/mozorg/home/includes/featured-vpn.html:11]: Typically we try and stick to putting CSS in the head. I don't think this is causing any problems here in newer browsers though. However the `page_css` block is meant for page level CSS, so using that block name here should probably be avoided.

I took a look, but I can't see any conflicting variables of selectors in `home-new.html`?

File: bedrock/mozorg/templates/mozorg/home/home-de.html
Comment (alexgibson) [inline bedrock/mozorg/templates/mozorg/home/home-de.html:23]: Can this `eu-home` bundle be removed? (looks like it's still hanging out in static-bundles.json)

File: bedrock/mozorg/templates/mozorg/home/home-contentful.html
Comment (alexgibson) [inline bedrock/mozorg/templates/mozorg/home/home-contentful.html:43]: Can this `home-pocket-promo` CSS bundle be removed too?

File: bedrock/mozorg/templates/mozorg/home/home-de.html
Comment (alexgibson) [inline bedrock/mozorg/templates/mozorg/home/home-de.html:27]: Another one (`ctd-promo`) that might be able to be removed?

**Review by alexgibson — CHANGES_REQUESTED** [review]
I noticed a few more old style sheets that might be able to be removed (the `home-eu` one also links to some old images in `/home/2018/`).

Other than that, everything looks great. Nice job on those functional tests! 💯

File: bedrock/mozorg/templates/mozorg/home/home-contentful.html
Comment (reemhamz) [reply bedrock/mozorg/templates/mozorg/home/home-contentful.html:43]: Spoke to Dan and he did mention to keep these promo files (Pocket promo and CTD promo) in the codebase in case they do need to come up again soon after launching the new homepage. In the meantime, we're not going to be using these promotions. We can remove them from our codebase a month or so after launch, if we don't face any hiccups.

**Comment by reemhamz** [comment]
@alexgibson 

> I took a look, but I can't see any conflicting variables of selectors in home-new.html?

Ah, seems like I was adding the VPN promo conditional switch for the CSS bundle within the same conditional block as the banners and what was causing a clash. I separated them and both seem to function well now.

File: bedrock/mozorg/templates/mozorg/home/includes/featured-vpn.html
Comment (reemhamz) [reply bedrock/mozorg/templates/mozorg/home/includes/featured-vpn.html]: Ah, seems like I was adding the VPN promo conditional switch for the CSS bundle within the same conditional block as the banners and what was causing a clash. I separated them and both seem to function well now.

**Comment by reemhamz** [comment]
Just a reminder that this switch needs to be approved for the promo to go live after this PR merges: https://github.com/mozilla/bedrock/pull/13447

**Review by alexgibson — CHANGES_REQUESTED** [review]
Updates to the new page look good!

The old home page is throwing an error though, since `{% include 'mozorg/home/includes/mr2-promo.html' %}` no longer exists. 

I'd suggest taking a look through the old template just to make sure we've removed references to those old promos that no longer exist.

**Review by alexgibson — APPROVED** [review]
r+

**Pull request merged**
